### DISCUSSION TITLE
Allow enum generation in addition to union types

### DISCUSSION THREAD

**Discussion by awdyson** [reduxjs/redux-toolkit#2144] [Ideas]
It'd be really convenient to be able to iterate over things like this:
```typescript
export type TimeZoneCode =
  | 'Africa/Abidjan'
  | 'Africa/Accra'
  | 'Africa/Addis_Ababa'
  | 'Africa/Algiers'
  | 'Africa/Asmara'
  | 'Africa/Bamako'
  | 'Africa/Bangui'
  | 'Africa/Banjul'
  | 'Africa/Bissau'
  // ...
```

**Comment by phryneas** [comment]
To be honest, without someone already having opened an issue on this (https://github.com/reduxjs/redux-toolkit/issues/1889 by the way), I would not have the slightest idea what you are talking about.
